pkg/daemon/server/service/isbs_query_service.go

```diff
@@ -29,28 +29,22 @@ func (is *isbSvcQueryService) ListBuffers(ctx context.Context, req *daemon.ListB
 	resp := new(daemon.ListBuffersResponse)
 
 	buffers := []*daemon.BufferInfo{}
-	for _, buffer := range is.pipeline.GetAllBuffers() {
-		if buffer.Type != v1alpha1.EdgeBuffer {
-			continue
-		}
-		bufferInfo, err := is.client.GetBufferInfo(ctx, buffer)
+	for _, edge := range is.pipeline.Spec.Edges {
+		buffer := v1alpha1.GenerateEdgeBufferName(is.pipeline.Namespace, is.pipeline.Name, edge.From, edge.To)
+		bufferInfo, err := is.client.GetBufferInfo(ctx, v1alpha1.Buffer{Name: buffer, Type: v1alpha1.EdgeBuffer})
 		if err != nil {
 			return nil, fmt.Errorf("failed to get information of buffer %q", buffer)
 		}
 		log.Debugf("Buffer %s has bufferInfo %+v", buffer, bufferInfo)
-		vFrom, vTo := is.pipeline.FindVerticesWithEdgeBuffer(buffer.Name)
-		if vFrom == nil || vTo == nil {
-			return nil, fmt.Errorf("buffer %q not found from the pipeline", buffer)
-		}
-		bufferLength, bufferUsageLimit := getVertexLimits(is.pipeline, vFrom)
+		bufferLength, bufferUsageLimit := getBufferLimits(is.pipeline, edge)
 		usage := float64(bufferInfo.TotalMessages) / float64(bufferLength)
 		if x := (float64(bufferInfo.PendingCount) + float64(bufferInfo.AckPendingCount)) / float64(bufferLength); x < usage {
 			usage = x
 		}
 		b := &daemon.BufferInfo{
 			Pipeline:         &is.pipeline.Name,
-			FromVertex:       &vFrom.Name,
-			ToVertex:         &vTo.Name,
+			FromVertex:       &edge.From,
```

Comment:
`FromVertex` pointing to `edge.From` data could be confusing (same for `To`)? Rename `FromVertex` to `FromEdge` ?

Reply:
Not edgeFrom and edgeTo?

If we want to rename, do it later together with UI change - it's only used in the UI.

Reply:
@Krithika3 fyi